### PR TITLE
docs: final observability edits

### PR DESCRIPTION
**Pull request by kkunapuli**
## Description

Add note on transformation of labels, in the case of workspace, username, or other pod labels containing special characters.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/integrations/observability/_index.rst [hunk 1]
```diff
@@ -273,15 +273,18 @@ Health Status
...
 
 Viewing Metrics
 ===============
 
 The Determined Master labels the pods it creates with specific state values. These pod labels can be
 accessed via the ``kube_pod_labels`` metric from kube-state-metrics. Label names will be formatted
-as ``label_determined_ai_<label_name>``, e.g. ``label_determined_ai_container_id``.
+as ``label_determined_ai_<label_name>``, e.g. ``label_determined_ai_container_id``. Kubernetes
+limits characters allowed in pod labels to alphanumeric characters, underscores, hyphens, and dots.
+Other characters in Determined resource names will be converted to an underscore, ``_``, before
```
Comment (NicholasBlaskey) [inline docs/integrations/observability/_index.rst:286]: what happens if the label is too long?

**Review by NicholasBlaskey — APPROVED** [review]
LGTM

File: docs/integrations/observability/_index.rst (hunk 1 above)
Comment (kkunapuli) [reply docs/integrations/observability/_index.rst:286]: The label is truncated if it's more than 63 characters it is truncated. Then, any non-alphanumeric characters at the end are removed ... so the final label can be less than 63 characters. 

63 characters is a lot; I don't think users would run into this very often. Do you think it's an important detail to include?

File: docs/integrations/observability/_index.rst [hunk 2]
```diff
@@ -273,15 +273,18 @@ Health Status
...
+being added as a pod label.
```
Comment (tara-det-ai) [inline docs/integrations/observability/_index.rst:287]: Viewing Metrics
===============

The Determined Master assigns specific state values to the pods it creates. These pod labels can be accessed via the `kube_pod_labels` metric from "kube-state-metrics". Label names are formatted as `label_determined_ai_<label_name>`, such as `label_determined_ai_container_id`.

Kubernetes restricts pod labels to alphanumeric characters, underscores, hyphens, and dots. Any other characters in Determined resource names will be converted underscores (`_`) before being added as a pod label.

**Review by tara-det-ai — APPROVED** [review]
suggested edits

File: docs/integrations/observability/_index.rst (hunk 1 above)
Comment (NicholasBlaskey) [reply docs/integrations/observability/_index.rst:286]: I was just curious. Might be good to include if someone does have long names for whatever reason

**Pull request merged**